<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Impact of tree choice in metagenomics differential abundance studies</title>
    <meta charset="utf-8" />
    <meta name="author" content="Antoine Bichat" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link href="libs/ionicons/css/ionicons.min.css" rel="stylesheet" />
    <link href="libs/font-awesome/css/fontawesome-all.min.css" rel="stylesheet" />
    <link rel="stylesheet" href="these.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">

class: title-slide, center, middle
count: false






&lt;br&gt;


# Influence du choix de l‚Äôarbre dans les √©tudes d‚Äôabondance diff√©rentielle

&lt;br&gt;

## Antoine Bichat
&lt;div style = "margin-top: -30px"&gt;&lt;/div&gt;
&lt;!-- ### <i class="ion  ion-social-twitter "></i> @_abichat --&gt;

### 6 juin 2019 - Journ√©es de Statistique - Nancy

&lt;br&gt;

#### En collaboration avec C. Ambroise (LaMME), M. Mariadassou (MaIAGE) &amp; J. Plassais (Enterome)


---
class: center, middle, inverse

# Context


---
## Microbiota


_Ecological community of microorganisms that reside in an environmental niche_ 

--

.footnote[üìò Gut: The Inside Story of Our Body's Most Underrated Organ  (Giulia Enders) &lt;br&gt; üìÑ Opstelten et al. (2016), Bokulich et al. (2016), Blander et al. (2017)]

.pull-left[

#### Some figures for human gut

* `\(10^{\small{14}}\)` bacterial cells in one gut...

* ... weighing 2 kg 

* More than 1&amp;#8239;500 different species 

* More than 10 millions unique genes


&lt;img src="index_files/figure-html/citations-1.png" width="288" style="display: block; margin: auto;" /&gt;

]

--


.pull-right[

#### Proven associations

* Immune system

* Crohn's disease

* Vaginosis

* Diabete

* Tobacco

* Diet

* Antibiotics

* Birth mode

]


---
## Data - abundances of taxa

.footnote[üìÑ Ravel et al. (2011)]


```
  # A tibble: 122 x 395
     Taxa     S001  S002  S003  S004  S005  S006  S007  S008  S009  S010
     &lt;chr&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
   1 Lactob‚Ä¶  2318  1388  1361  2256    88  1770  1490   119  2136  1790
   2 Prevot‚Ä¶     0     1     1     0   525     7   134   753     0     0
   3 Megasp‚Ä¶     0     1     0     0   402     0     4   102     0     0
   4 Sneath‚Ä¶     0     0     0     0   302     0    35   272     0     0
   5 Atopob‚Ä¶     0     1     0     0    84     0    12    54     0     0
   6 Strept‚Ä¶     0     0     3     0     0     0   138     4     0     2
   7 Dialis‚Ä¶     0     1     0     0   152     4     2   192     0     0
   8 Anaero‚Ä¶     0     1     3     2     0     9    12    13     0     0
   9 Pepton‚Ä¶     0     1     0     0     7     2     6    50     0     0
  10 Eggert‚Ä¶     0     0     0     0     2     0     0     7     0     0
  # ‚Ä¶ with 112 more rows, and 384 more variables
```

--

* Count data (or compositional) data

* Zero-inflated data

* Correlation between species

* Counts spanning several orders of magnitude: `\(1 \rightarrow 10^{\small{8}}\)`


---
## Data - taxonomy

.footnote[üìÑ Ravel et al. (2011)]


```
  # A tibble: 129 x 5
    Phylum         Class          Order          Family         Genus        
    &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;        
  1 Actinobacteria Actinobacteria Actinomycetal‚Ä¶ Actinomycetac‚Ä¶ Actinobaculum
  2 Actinobacteria Actinobacteria Actinomycetal‚Ä¶ Actinomycetac‚Ä¶ Actinomyces  
  3 Actinobacteria Actinobacteria Actinomycetal‚Ä¶ Actinomycetac‚Ä¶ Arcanobacter‚Ä¶
  4 Actinobacteria Actinobacteria Actinomycetal‚Ä¶ Actinomycetac‚Ä¶ Mobiluncus   
  5 Actinobacteria Actinobacteria Actinomycetal‚Ä¶ Actinomycetac‚Ä¶ Varibaculum  
  # ‚Ä¶ with 124 more rows
```


---
class: nologo
count: false
## Data - taxonomy


```
  # A tibble: 129 x 5
    Phylum         Class          Order          Family         Genus        
    &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;        
  1 Actinobacteria Actinobacteria Actinomycetal‚Ä¶ Actinomycetac‚Ä¶ Actinobaculum
  2 Actinobacteria Actinobacteria Actinomycetal‚Ä¶ Actinomycetac‚Ä¶ Actinomyces  
  3 Actinobacteria Actinobacteria Actinomycetal‚Ä¶ Actinomycetac‚Ä¶ Arcanobacter‚Ä¶
  4 Actinobacteria Actinobacteria Actinomycetal‚Ä¶ Actinomycetac‚Ä¶ Mobiluncus   
  5 Actinobacteria Actinobacteria Actinomycetal‚Ä¶ Actinomycetac‚Ä¶ Varibaculum  
  # ‚Ä¶ with 124 more rows
```

.center[&lt;img src="img/tree_tax.png"/ width="500"&gt;]

---
class: center, middle, inverse

# Differential abundance studies

---
## Statistical issue

Which taxa are differentially abundant between groups? 

--

* ANOVA / GLM

* Wilcoxon rank sum test

--

&lt;br&gt;

But hundreds of taxa in a single study...

Need for a controling procedure!

* Bonferroni (FWER)

---
count: false
## Statistical issue

Which taxa are differentially abundant between groups? 

* ANOVA / GLM

* Wilcoxon rank sum test

&lt;br&gt;

But hundreds of taxa in a single study...

Need for a controling procedure!

* ~~Bonferroni (FWER)~~ `\(\rightarrow\)` too conservative

--

* Benjamini-Hochberg (FDR)

--

&lt;br&gt;

üí™ Is it possible to do better?


---
## Incorporate taxonomy in the analysis?

.footnote[üìÑ Philippot et al. (2010), Koeppel &amp; Wu (2012)]

* Taxonomy is a proxy for the structure of the data


* Coherent with ecological niches


* Independant from count data and metadata

&lt;br&gt;

.center[
  &lt;img src="img/coherence.png" width="550"&gt;
]

---
## Hierarchical FDR

.footnote[üìÑ Yekutieli (2008)]

`\(\mathcal{T}_t = \left\{H_i \mid \text{Par}(i) = t\right\}\)`

Descending method:

* Test the family `\(\mathcal{T}_0\)`

* If node `\(t\)` is rejected, test `\(\mathcal{T}_t\)` with a classical BH procedure at level `\(q\)`

&lt;br&gt;

.center[
  &lt;img src="img/tree_hyp.png" width="420"&gt;
]

---
count: false
## Hierarchical FDR

.footnote[üìÑ Yekutieli (2008)]

`\(\mathcal{T}_t = \left\{H_i \mid \text{Par}(i) = t\right\}\)`

Descending method:

* Test the family `\(\mathcal{T}_0\)`

* If node `\(t\)` is rejected, test `\(\mathcal{T}_t\)` with a classical BH procedure at level `\(q\)`

&lt;br&gt;

.center[
  &lt;img src="img/tree_hyp_1.png" width="420"&gt;
]

---
count: false

## Hierarchical FDR

.footnote[üìÑ Yekutieli (2008)]

`\(\mathcal{T}_t = \left\{H_i \mid \text{Par}(i) = t\right\}\)`

Descending method:

* Test the family `\(\mathcal{T}_0\)`

* If node `\(t\)` is rejected, test `\(\mathcal{T}_t\)` with a classical BH procedure at level `\(q\)`

&lt;br&gt;


.center[
  &lt;img src="img/tree_hyp_2.png" width="420"&gt;
]

---
count: false
## Hierarchical FDR

.footnote[üìÑ Yekutieli (2008)]

`\(\mathcal{T}_t = \left\{H_i \mid \text{Par}(i) = t\right\}\)`

Descending method:

* Test the family `\(\mathcal{T}_0\)`

* If node `\(t\)` is rejected, test `\(\mathcal{T}_t\)` with a classical BH procedure at level `\(q\)`

&lt;br&gt;

.center[
  &lt;img src="img/tree_hyp_3.png" width="420"&gt;
]

--

&lt;br&gt;

.center[
__This procedure controls the FDR at level__ 

`\(\large{1.44 \times q \times \frac{\#\text{discoveries } +\text{ } \#\text{families tested}}{\#\text{discoveries } +\text{ } 1}}\)`
]


---
## Bayesian smoothing

.footnote[üìÑ Xiao et al. (2017)]

Denote by `\(\mathbf{z}\)` the vector of observed z-values and `\(\mathbf{\mu}\)` the vector of "true" z-values

--

&lt;br&gt;

Assume that 
`\(\mathbf{z} | \mathbb{\mu} \sim \mathcal{N}_n \left( \mathbb{\mu}, \sigma^2 \mathbf{I}_m \right)\)` and `\(\mathbf{\mu} \sim \mathcal{N}_m\left(\gamma \mathbf{1} , \tau^2 \mathbf{C}_{\rho} \right)\)` 

--

then 
`$$\mathbf{z} \sim \mathcal{N}_m \left(\gamma \mathbf{1},\tau^2 \mathbf{C}_{\rho} + \sigma^2 \mathbf{I}_m\right)$$`

and Bayes formula gives 
 
`$$\mathbb{\mu}^* = \left(\mathbf{I}_m + \frac{\sigma_0^2}{\tau_0^2} \mathbf{C}_{\rho_0}^{-1}\right)^{-1}\left(\frac{\sigma_0^2}{\tau_0^2} \mathbf{C}_{\rho_0}^{-1}\gamma_0 \mathbf{1} + \mathbf{z}\right)$$`

with `\(\sigma_0\)`, `\(\tau_0\)`, `\(\rho_0\)` and `\(\gamma_0\)` hypermarameters

--

&lt;br&gt;

After smoothing, a multiple testing correction could be done on smoothed values


---
## Which tree?


#### Taxonomy?

* Proxy for correlations at high-level niches

* Not so much for low-level niches?

* Not available everytime


---
class: nologo
count: false
## Which tree?


#### Taxonomy?

* Proxy for correlations at high-level niches

* Not so much for low-level niches?

* Not available everytime

#### Correlation tree?

* Actual correlation between taxa

* Computed from data using pairwise correlation


.center[
&lt;img src="img/tree_cor.png" width="350"&gt;
]

---
class: center, middle, inverse

# Comparison of trees

---
## Billera-Holmes-Vogtamnn distance

.footnote[üìÑ Billera, Holmes &amp; Vogtmann (2001)]

The BHV distance is the length of the unique shortest path between the trees on treespace

.pull-left[
  .center[
&lt;img src="img/dBHV_2.png" height=300/&gt;
  ]
]

.pull-right[
  .center[
    &lt;img src="img/dBHV_4.png" height=300/&gt;
  ]
]


---
## Quantifying distance between trees


* __trees of primary interest__

  * &lt;span style="color:#C77CFF"&gt;correlation tree on original data&lt;/span&gt;
  
  * &lt;span style="color:#F8766D"&gt;taxonomy&lt;/span&gt;
  
--

* __what is the confident region for the correlation tree?__

--

  * &lt;span style="color:#00BFC4"&gt;correlation trees on boostrapped data (resampling on samples)&lt;/span&gt;

--

* __are trees significantly closer than two random trees?__

--

  * &lt;span style="color:#7CAE00"&gt;trees created by random shuffling of correlation tree tip labels&lt;/span&gt;
  
  * &lt;span style="color:#FFA500"&gt;trees created by random shuffling of taxonomy tip labels&lt;/span&gt;

--

&lt;br&gt;

We compute all pairwise distances between these trees

---
## Random shuffling

.center[&lt;img src="img/shuffling.png", height="500"&gt;]

---
## Dataset

.footnote[üìÑ Ravel et al. (2011)]

* Vaginal microbiome of non pregnant women sequenced by 16S

* 40 different genera after filtering (~&amp;#8239;30&amp;#8239;%)

&lt;br&gt;

.pull-left[ 
  &lt;img src="img/tree_tax.png"&gt;
  .center[Taxonomy]
]

.pull-right[
  &lt;img src="img/tree_cor.png"&gt;
  .center[Correlation tree] 
]

---
## Pairwise distances

.pull-left[
.center[&lt;img src="img/prez_ravel_comp_box.png" width="650"&gt;]
]

--

.pull-right[
.center[&lt;img src="img/prez_ravel_comp_pcoa.png" width="650"&gt;]
]

--

&lt;br&gt;

üòÄ The correlation tree is different from the taxonomy


---
class: center, middle, inverse

# Evaluation of hFDR

---
## Dataset

.footnote[üìÑ Caporaso et al. (2011), Sankaran &amp; Holmes (2014)]

* Small subset of the `GlobalPatterns` dataset narrowed to Chlamydiae phylum

* 21 different OTUs

* 26 samples representing 9 very different environments: soil, ocean, feces, skin...

--

## Method

* Find which bacteria are differentially abundant between environments

* Association using Fisher statistic (ANOVA)

* Correction with hierarchical FDR

---
## Abundances of detected species

.center[&lt;img src="img/realdata_chlamydia_box.png" height=500/&gt;]


---
## Representation of evidences on trees

.center[&lt;img src="img/tree_mirors.png" height=500/&gt;]

---
## But...

`\(\alpha = 0.05\)` is only the family-level FDR.

The _a posteriori_ global FDR is:

* `\(\alpha' = 0.32\)` for phylogenetic correction

* `\(\alpha' = 0.324\)` for correlation correction


--

A BH procedure at the same global FDR level leads to 15 discoveries (+5)

--

&lt;br&gt;

üòÄ Using correlation tree instead of taxonomy yields more results


‚òπÔ∏è Vanilla FDR beats hFDR for a given level



---
class: center, middle, inverse

# Evaluation of z-scores smoothing


---
## Dataset

.footnote[üìÑ Zeller et al. (2014)]

* Dataset from cancer study

* 119 different genera (after filtering)

* 199 samples: 42 adenoma, 91 carcinoma and 66 control

--

## Method

* Find which bacteria are differentially abundant between diseases

* Association using Kruskal-Wallis test

* Correction with hierarchical p-value smoothing


---
## Impact of the tree

.center[&lt;img src="img/zeller-lines-bold.png" height=430/&gt;]

--

üòä z-scores smoothing is slightly better than vanilla FDR

‚òπÔ∏è All hierachies give highly similar results

---
class: center, middle, inverse

# Simulations

---
## Workflow

.footnote[üìÑ Brito et al. (2016)]

* Simulate DA taxa starting from an homogeneous dataset

* Correction with BH and hierarchical p-value smoothing


.center[&lt;img src="img/workflow_simu.png" height=450/&gt;]

---
## Evaluation

.center[&lt;img src="img/simu_np_lines_prez.png" height=390/&gt;]

--

üòÄ Using correlation tree instead of taxonomy yields more results

‚òπÔ∏è Vanilla BH is better 

ü§î Taxonomy is worst than random trees 


---
class: center, middle, inverse

# Conclusions

---

## Conclusions 


üòÄ Correlation tree and taxonomy are very different

üòÄ Replacing taxonomy with correlationn increases the TPR

--

&lt;br&gt;

‚òπÔ∏è Vanilla BH is more powerful than hFDR

‚òπÔ∏è Bayesian smoothing does not really depend on the tree for z-scores smoothing

--

&lt;br&gt;


üì¶ `correlationtree`

üì¶ `yatah` <i class="fab  fa-github "></i>

üì¶ `evabic` <i class="fab  fa-github "></i>


---
class: title-slide, center, middle
count: false

&lt;br&gt;

# Merci pour votre attention !

&lt;br&gt;

# Des questions ?


&lt;br&gt;

#### <i class="fas  fa-link "></i> &lt;a href="https://abichat.github.io" target="_blank"&gt;abichat.github.io&lt;/a&gt; &amp;emsp; <i class="fab  fa-github "></i>  &lt;a href="https://github.com/abichat" target="_blank"&gt;@abichat&lt;/a&gt; &amp;emsp; <i class="fab  fa-twitter "></i> &lt;a href="https://twitter.com/_abichat" target="_blank"&gt;@_abichat&lt;/a&gt;
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false,
"slideNumberFormat": "%current% / %total%"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
